src/identity/core/getName.ts

```diff
@@ -1,15 +1,53 @@
-import { publicClient } from '../../network/client';
+import { mainnet } from 'viem/chains';
 import type { GetName, GetNameReturnType } from '../types';
+import L2ResolverAbi from '../abis/L2ResolverAbi';
+import {
+  baseChainsIds,
+  chainsById,
+  ensUniversalResolverChainIds,
+  getChainPublicClient,
+} from '../../network/getChainPublicClient';
+import { resolverAddressesByChainId } from '../constants';
+import { convertReverseNodeToBytes } from './convertReverseNodeToBytes';
 
 /**
  * An asynchronous function to fetch the Ethereum Name Service (ENS)
  * name for a given Ethereum address. It returns the ENS name if it exists,
  * or null if it doesn't or in case of an error.
  */
+
 export const getName = async ({
   address,
+  chainId = mainnet.id,
 }: GetName): Promise<GetNameReturnType> => {
-  const ensName = await publicClient.getEnsName({
+  const chainSupportsUniversalResolver =
+    ensUniversalResolverChainIds.includes(chainId);
+  const chainIsBase = baseChainsIds.includes(chainId);
```

Comment:
this should be his own function, as `isBaseChain()` and live at top level of OnchainKit.